integration-tests/src/test/java/com/datastax/oss/driver/api/core/loadbalancing/NodeTargetingTestIT.java

```diff
@@ -0,0 +1,115 @@
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@Category(ParallelizableTests.class)
+public class NodeTargetingTestIT {
+
+  @Rule public SimulacronRule simulacron = new SimulacronRule(ClusterSpec.builder().withNodes(5));
+
+  @Rule public SessionRule<CqlSession> sessionRule = SessionRule.builder(simulacron).build();
+
+  @Before
+  public void clear() {
+    simulacron.cluster().clearLogs();
+    simulacron.cluster().clearPrimes(true);
+  }
+
+  @Test
+  public void should_use_host_on_statement() {
+    Collection<Node> nodeCol = sessionRule.session().getMetadata().getNodes().values();
+    List<Node> nodes = new ArrayList(nodeCol);
+    for (int i = 0; i < 10; i++) {
+      int hostIndex = i % 4 + 1;
+      Node node = nodes.get(hostIndex);
+
+      // given a statement with host explicitly set.
+      Statement statement = SimpleStatement.newInstance("select * system.local").setNode(node);
+
+      // when statement is executed
+      ResultSet result = sessionRule.session().execute(statement);
+
+      Node coordinator = result.getExecutionInfo().getCoordinator();
+      // then the query should have been sent to the configured host.
+      assertThat(result.getExecutionInfo().getCoordinator()).isEqualTo(node);
+    }
+  }
+
+  @Test
+  public void should_fail_if_host_fails_query() {
+    String query = "mock";
+    Collection<Node> nodeCol = sessionRule.session().getMetadata().getNodes().values();
+    List<Node> nodes = new ArrayList(nodeCol);
+    simulacron.cluster().node(3).prime(when(query).then(unavailable(ConsistencyLevel.ALL, 1, 0)));
+
+    // given a statement with a host configured to fail the given query.
+    Node node1 = nodes.get(3);
+    Statement statement = SimpleStatement.newInstance(query).setNode(node1);
+    // when statement is executed an error should be raised.
+    try {
+      ResultSet result = sessionRule.session().execute(statement);
+      fail("Should have thrown NoNodeAvailableException");
+    } catch (AllNodesFailedException e) {
+      assertThat(e.getErrors().size()).isEqualTo(1);
+      assertThat(e.getErrors().get(node1)).isInstanceOf(UnavailableException.class);
+    }
+  }
+
+  @Test
+  public void should_fail_if_host_is_not_connected() {
+    // given a statement with host explicitly set that for which we have no active pool.
+    simulacron.cluster().node(4).close();
+    ;
+    Collection<Node> nodeCol = sessionRule.session().getMetadata().getNodes().values();
+    List<Node> nodes = new ArrayList(nodeCol);
+    Node node4 = nodes.get(4);
+    Statement statement = SimpleStatement.newInstance("select * system.local").setNode(node4);
+    try {
+      // when statement is executed
+      sessionRule.session().execute(statement);
+      fail("Query should have failed");
+    } catch (AllNodesFailedException e) {
```

Comment:
More precisely, it's a `NoNodeAvailableException` (a subclass).

Which IMO is also more clear for the end user: "no node was available" and you targeted a node => the targeted node was not available.